packages/n4s/src/rules/isKeyOf.ts

```diff
@@ -0,0 +1,13 @@
+import bindNot from 'bindNot';
+import { isBoolean } from 'isBooleanValue';
+import { isNullish } from 'isNullish';
+import { isStringValue } from 'isStringValue';
+
+export function isKeyOf(key: string | symbol | number, obj: any): boolean {
+  const shouldTerminate =
+    isNullish(obj) || isBoolean(obj) || isStringValue(obj);
+
+  return shouldTerminate ? false : key in obj;
```

Comment:
`shouldTerminate` doesn't cover all cases, for example - when the passed object is a `Symbol`.

I think it is enough to assume good intent, and follow the happy path here, thus not needing to validate the object. A few levels up the stack `enforce` actually does a try/catch, and errors thrown are treated as false.

I would simplify it like that:

```suggestion
export function isKeyOf(key: string | symbol | number, obj: any): boolean {
  return key in obj;
```

This might also cause some of the tests to fail, but then you just have to check that this function throws.